app/todolist.py

```diff
@@ -19,12 +25,19 @@ def set_item(self,item:Item):
             raise TypeError()
         if self.verify_updated_at() is False:
             raise ValueError("Waiting 30 min")
+        
         if self.verify_length() is False:
             raise ValueError("You have reached the limit of items in your list")
-        if item in self.__items:
+
+        if any(_item for _item in self.__items if item.get_name() == _item.get_name()):
             raise ValueError("Item already exists")
```

Comment:
Ici je test si y a pas d'item dans la liste qui a le même name, parce que en faisant `item in self.__items` ça va juste vérifier la référence. Et vu que tu ajoutes 2 fois la même variable ça compte comme double.

Si je crée 2 Item identiques sur 2 variables, ta condition sera true alors que c'est les mêmes items. Je sais pas si c'est clair ou pas

Reply:
Oui c'est clair j'ai compris

Reply:
Pourquoi avoir ajouté un owner dans les attributs de ta class **Todolist** ?
C'est un **User** qui a une **TodoList**

Reply:
C'est pour lui envoyer un email en son nom

Reply:
Humm je vois mais vu que tu raise juste exception une **send_email** sa devrait suffire

Reply:
Je vérifie si le mail a été envoyé à la bonne personne aussi

Reply:
Ok j'avais pas vu dans ton test